features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/Base64Serializer.kt

```diff
@@ -35,11 +39,18 @@ internal class Base64Serializer private constructor(
     private val base64LRUCache: Cache<Drawable, CacheData>,
     private val bitmapPool: BitmapPool?,
     private val logger: InternalLogger,
-    private val md5HashGenerator: MD5HashGenerator
+    private val md5HashGenerator: MD5HashGenerator,
+    private val recordedDataQueueHandler: DataQueueHandler,
+    private val applicationId: String
 ) {
     private var isBase64CacheRegisteredForCallbacks: Boolean = false
     private var isBitmapPoolRegisteredForCallbacks: Boolean = false
 
+    // resources previously sent in this session -
+    // optimization to avoid sending the same resource multiple times
+    private val previouslySentResources: MutableSet<String> =
```

Comment:
we can call this one simply `resourcesSent` (or maybe `resourcesSeen`)

```suggestion
    private val resourcesSent: MutableSet<String> =
```

can we also do the napkin math for this one? if I understand it correctly, value will be MD5 hash, which is 32 characters. If Android VM encoding used is UTF-16, than 2 bytes per character, meaning 64 bytes roughly (not counting service fields of the String object) per item. How many items do we expect here?

Reply:
Probably we can remove this set and use the existing lrucache - if an item is in the lrucache we know we've seen it before. The disadvantage only being that an item may be evicted from the lru and we would send it again

Reply:
As many images as there are in the session. Perhaps we could take advantage of the existing lrucache to make the memory usage bounded, although since multiple drawables could have the same resourceId it would need to be something like a map of resourceIds to refcount, and only remove when we see no more references for the resourceId

Reply:
It is fine to keep this as well if we expect the total size to be small, that is why I'm curious about the estimations. Let's say we sent 1000 resources, each ID string takes, for example, 80 bytes in the memory. This is 80 Kb, which is okay-ish.